weave/wandb_interface/wandb_stream_table.py

```diff
@@ -136,57 +63,78 @@ def __init__(
-            )
-        if not hasattr(self, "_artifact"):
-            uri = runfiles_wandb.WeaveWBRunFilesURI.from_run_identifiers(
-                self._entity_name,
-                self._project_name,
+    def _ensure_remote_initialized(self) -> None:
+        with self._lock:
+            self._lite_run.ensure_run()
+            print_url = False
+            self._artifact = self._stream_table_artifact()
+            if print_url:
+                base_url = wandb_public_api().settings["base_url"]
+                if base_url.endswith("api.wandb.ai"):
+                    base_url = base_url.replace("api", "weave.")
+                url = f"{base_url}/?exp=get%28%0A++++%22wandb-artifact%3A%2F%2F%2F{self._entity_name}%2F{self._project_name}%2F{self._table_name}%3Alatest%2Fobj%22%29%0A++.rows"
+                printer = get_printer(_get_python_type() != "python")
+                printer.display(
+                    f'{printer.emoji("star")} View data at {printer.link(url)}'
+                )
+
+    def _stream_table_artifact(self) -> "Artifact":
+        if self._artifact is None:
+            self._artifact = Artifact(
                 self._table_name,
+                type="stream_table",
+                metadata={
+                    "_weave_meta": {
+                        "is_panel": False,
+                        "is_weave_obj": True,
+                        "type_name": "stream_table",
+                    },
+                },
             )
-            self._artifact = WandbLiveRunFiles(name=uri.name, uri=uri)
-            self._artifact.set_file_pusher(self._lite_run.pusher)
-        if print_url:
-            base_url = environment.weave_server_url()
-            url = f"{base_url}/?exp=get%28%0A++++%22wandb-artifact%3A%2F%2F%2F{self._entity_name}%2F{self._project_name}%2F{self._table_name}%3Alatest%2Fobj%22%29%0A++.rows"
-            printer = get_printer(_get_python_type() != "python")
-            printer.display(f'{printer.emoji("star")} View data at {printer.link(url)}')
-        return self._weave_stream_table
+            with self._artifact.new_file("obj.object.json") as f:
+                payload = {
+                    "_type": "stream_table",
+                    "table_name": self._table_name,
+                    "project_name": self._project_name,
+                    "entity_name": self._entity_name,
+                }
+                f.write(json.dumps(payload))
+            with self._artifact.new_file("obj.type.json") as f:
+                payload = {
+                    "type": "stream_table",
+                    "_base_type": {"type": "Object"},  # type: ignore[dict-item]
+                    "_is_object": True,  # type: ignore[dict-item]
+                    "table_name": "string",
+                    "project_name": "string",
+                    "entity_name": "string",
+                }
+                f.write(json.dumps(payload))
+            self._lite_run.log_artifact(self._artifact)
```

Comment:
A part of me wants to put this into gorilla when a run is flagged as "stream_table".  The advantage being we could change the schema of new stream tables without requiring SDK updates.

Reply:
i think in the first 1-2 months we have it here, but i agree we should move to gorilla. One huge benefit is if we have a standard way of initializing, it can be used by the proxy and this commonly. In fact, if that was the case, the StreamTable class might be able to entirely disappear

Reply:
or at least most of the business logic would be no-ops